src/services/uniswapV3/UniswapV3Service.ts

```diff
@@ -0,0 +1,264 @@
+import {injectable} from 'inversify';
+import {Token, CurrencyAmount} from '@uniswap/sdk-core';
+import {FeeAmount, Pool, TickMath, tickToPrice} from '@uniswap/v3-sdk';
+import JSBI from 'jsbi';
+
+import {GraphTick, BarChartTick, TickProcessed} from './interfaces.js';
+import axios from 'axios';
+
+enum Direction {
+  ASC,
+  DESC,
+}
+
+const MAX_INT128 = JSBI.subtract(JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(128)), JSBI.BigInt(1));
+
+@injectable()
+export default class UniswapV3Service {
```

Comment:
what is this service for? let's use names that means something

Reply:
in general each service should have one entry point `apply`, anythign else should be either private or moved to separate service